### PR TITLE
Fix incidents

### LINKED ISSUE #102: Fix inserting incidents into the database
### Task

- [x] Correct transform_national.py to populate end and start dates if not given from the NationalRail API, to ensure all incidents are being inserted into the database.
- [x] If no ```incident_end``` date is given, use ```incident_start``` date.
- [x] If no ```incident_start``` date then use ```creation_time```. 
- [x] Allow nullable on columns ````is_planned```, ```incident_summary```, ```incident_description```, ```incident_uri```, ```affected_routes```. Use ALTER TABLE COLUMN. 

### User Story
As a developer, I must make sure all incidents are inserted into the database to ensure that users are notified for alerts accurately.

### PR DESCRIPTION
**Pull request by FarihaChoudhury**
Populates data for incidents without a start or end time so that they can be inserted into the database with the most accuracy possible. Incidents without a start time have the start time set to the creation time. Incidents without end times have the end time set to the start time.

Alter incident table in the schema with NOT NULL in certain columns. 

Closes #102

### CODE REVIEW

File: database/schema.sql [hunk 1]
```diff
@@ -15,11 +15,11 @@ CREATE TABLE incident(
...
-    incident_summary TEXT NOT NULL,
-    incident_description TEXT NOT NULL,
-    incident_uri TEXT NOT NULL,
-    affected_routes TEXT NOT NULL
+    is_planned BOOLEAN,
+    incident_summary TEXT,
+    incident_description TEXT,
+    incident_uri TEXT,
+    affected_routes TEXT
 );
```
Comment (Zhi-704) [inline database/schema.sql:23]: Makes sense

File: national_rail/transform_national.py [hunk 2]
```diff
@@ -103,6 +103,11 @@ def process_pt_incidents(incidents: list[ET.Element], namespaces: dict) -> list[
         end_time = find_text_element(
             incident, 'ns:ValidityPeriod/com:EndTime', namespaces)
 
+        if not start_time:
+            start_time = creation_time
+        if not end_time:
+            end_time = start_time
+
```
Comment (Zhi-704) [inline national_rail/transform_national.py:110]: Good check

File: pdf_report/load_pdf.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+from dotenv import load_dotenv
+from boto3 import client
+from botocore.exceptions import NoCredentialsError
+import botocore.exceptions
+from botocore.exceptions import ClientError
+from psycopg2 import connect, Error as psycopg2_error
+from psycopg2.extensions import connection, cursor
+from psycopg2.extras import RealDictCursor
+
```
Comment (Zhi-704) [inline pdf_report/load_pdf.py:19]: Is this supposed to be in the pull request?

File: database/schema.sql (hunk 1 above)
Comment (A-Faris) [reply database/schema.sql:23]: I've updated the ERD

File: pdf_report/README.md [hunk 4]
```diff
@@ -1,2 +1,21 @@
...
+```text
+AWS_ACCESS_KEY=your_aws_access_key
+AWS_SECRET_KEY=your_aws_secret_access_key
+DB_PASSWORD=your_database_password
+DB_USERNAME=your_database_username
+DB_NAME=your_database_name
+DB_PORT=your_database_port
+DB_IP=your_database_ip_or_hostname
+SOURCE_EMAIL=your_email_to_send_ses
+S3_BUCKET_NAME=your_s3_bucket_name
```
Comment (A-Faris) [inline pdf_report/README.md:18]:
```suggestion
AWS_ACCESS_KEY=your_aws_access_key
AWS_SECRET_KEY=your_aws_secret_access_key

DB_PASSWORD=your_database_password
DB_USERNAME=your_database_username
DB_NAME=your_database_name
DB_PORT=your_database_port
DB_IP=your_database_ip_or_hostname

SOURCE_EMAIL=your_email_to_send_ses
S3_BUCKET_NAME=your_s3_bucket_name 
```

File: pdf_report/load_pdf.py (hunk 3 above)
Comment (FarihaChoudhury) [reply pdf_report/load_pdf.py:19]: Thank you, good spot

**Pull request closed without merging**